server.js

```diff
@@ -985,6 +984,48 @@ routes['join_room'] = function (user, data) {
     room.sync(user);
 }
 
+routes['search_youtube'] = function (user, data) {
+	if(user.room === null) return;
+	var room = user.room;
+
+	var query = encodeURIComponent(data.query);
+
+
+	
+	if (typeof(data.query !== query) || query.length < 1 || query.length > 100) {
```

Comment:
Please look up how typeof works. And also need to check data.query length instead of query length as URL encoding increases the length of the string. Also please move the constant of 100 to the top of the file.